src/gui/KM_GUIMenuOptions.pas

```diff
@@ -556,45 +528,22 @@ procedure TKMMenuOptions.BackClick(Sender: TObject);
 procedure TKMMenuOptions.RefreshKeyList;
 var
   I, prevI: Integer;
+  K: TKMKeyArea;
 begin
   prevI := ColumnBox_Options_Keys.TopIndex;
 
   ColumnBox_Options_Keys.Clear;
-
-  // Hide the debug keys
-  for I := 0 to gResKeys.Count - 5 do
-    ColumnBox_Options_Keys.AddItem(MakeListRow([gResKeys.GetFunctionNameById(I), gResKeys.GetKeyNameById(I)],
-                                               [$FFFFFFFF, $FFFFFFFF], [$FF0000FF, $FF0000FF]));
-
-  ColumnBox_Options_Keys.TopIndex := prevI;
-end;
-
-
-procedure TKMMenuOptions.RefreshKeyList2;
-var
-  I, D: Integer;
-begin
-  for I := 100 to 113 do
+  for K := Low(TKMKeyArea) to High(TKMKeyArea) do
   begin
-    case I of
-      100: D := 13;   // MapEdit Extra's menu
-      101: D := 112;  // MapEdit Terain Editing
-      102: D := 113;  // MapEdit Village Planning
-      103: D := 114;  // MapEdit Visual Scripts
-      104: D := 115;  // MapEdit Global Scripting
-      105: D := 116;  // MapEdit Main Menu
-      106: D := 49;   // MapEdit Sub-menu 1
-      107: D := 50;   // MapEdit Sub-menu 2
-      108: D := 51;   // MapEdit Sub-menu 3
-      109: D := 52;   // MapEdit Sub-menu 4
-      110: D := 53;   // MapEdit Sub-menu 5
-      111: D := 54;   // MapEdit Sub-menu 6
-      112: D := 121;  // Unassignable F10 key, Delphi special key
-      113: D := 122;  // Unassignable F11 key, debug menu
-    end;
-
-    ColumnBox_Options_Special_Keys.AddItem(MakeListRow([gResKeys.GetFunctionNameById(I), gResKeys.GetKeyName(D)], [$FFFFFFFF, $FFFFFFFF]));
+    ColumnBox_Options_Keys.AddItem(MakeListRow([gResTexts[KEY_SEP_TX[K]], ' '], [$FF3BB5CF, $FF3BB5CF], [$FF0000FF, $FF0000FF], -1));
+    // Hide the debug keys
```

Comment:
You are not hiding them, you are not showing them ;-)
Comment should read "Do not show debug keys"